File: 274/js/274.js
/**
 * @param {number[]} citations
 * @return {number}
 */
var hIndex = function(citations) {
    // sort first 
    citations.sort(function(a,b){return b - a})

    hId = 0
    for (const cited of citations) {
        if (hId+1 > cited) {
            return hId
        }
        hId++
    }

    hId = citations.length

    return hId
}

File: 274/py/274.py
class Solution:
    def hIndex(self, citations: List[int]) -> int:
        citations.sort(reverse=True)

        hId = 0
        i = 0
        for cited in citations :
            if i+1 > cited :
                hId = i
                return hId
            i = i+1

        hId = len(citations)

        return hId
